### ISSUE TITLE
Poem crash during f64 parsing

### ISSUE THREAD

**Issue by rxdiscovery** [poem-web/poem#883]
Hello,

Recently, an application made with Poem crashed when sending JSON responses.

I investigated and found the cause:

`thread 'tokio-runtime-worker' panicked at /home/XXXXX/.cargo/registry/src/index.crates.io-6f17d22XXXXXXX/poem-openapi-5.1.0/src/types/external/floats.rs:91:1:`

It is this macro that is behind the exception :
```rust
impl_type_for_floats!((f32, "float"), (f64, "double"));
```

analyzing the code generated by this macro, I find the culprit:

``` rust
       impl ToJSON for $ty {
            fn to_json(&self) -> Option<Value> {
                Some(Value::Number(Number::from_f64(*self as f64).unwrap())) //<----- UNWRAP <<<<
            }
        }

```
there's an “unwrap()” in the code (src/types/external/floats.rs), the exception is not handled.

In a post about a year ago, I pointed out the presence of several unwrap() functions in the Poem project source code. The unwrap() / except() functions should never be used in production code.

several files in **src/types/external/** contain unwrap() :
- array.rs
- bson.rs
- float.rs
- time.rs

Thank you for all your efforts behind this framework. :+1: 

## Specifications

- Version: 3.1.0

**Comment by sunli829** [comment]
Thanks for the bug report, I fixed it except for `array.rs`. Since I [`checked the length`](https://github.com/poem-web/poem/blob/f81424dedeec44cbdf4dbe2e828b0d3bbf579486/poem-openapi/src/types/external/array.rs#L50), the try_into there couldn't possibly fail.

**Comment by sunli829** [comment]
released in `v5.1.1`

**Comment by rxdiscovery** [comment]
@sunli829  Thank you very much for your responsiveness :1st_place_medal:
